### PR TITLE
Fixes card-24: rename action object from sock to zmq

### PR DESCRIPTION
**Pull request by revathskumar**
**Review only after merging #6**

This is as per the discussion happened in last review.

### CODE REVIEW

File: app/redux/store.js [hunk 1]
```diff
@@ -31,28 +31,31 @@ const zmqMiddlewareCreator = (zmq, subzmq) => {
...
           break;
         case 'DISCONNECT':
-          subzmq.unsubscribe();
+          subzmq.unsubscribe('');
 
-          zmq.disconnect();
-          subzmq.disconnect();
+          reqzmq.disconnect('tcp://localhost:5556');
+          subzmq.disconnect('tcp://localhost:5555');
           break;
```
Comment (siddharthbhagwan) [inline app/redux/store.js]: Could `localhost:5556` be moved to a constant, or some config file? Hardcoding directly where its used seems weird

File: app/redux/store.js [hunk 2]
```diff
@@ -31,28 +31,31 @@ const zmqMiddlewareCreator = (zmq, subzmq) => {
 
       switch (type) {
         case 'CONNECT':
-          zmq.connect(`tcp://${sock.ip || 'localhost'}:5556`);
-          subzmq.connect(`tcp://${sock.ip || 'localhost'}:5555`);
+          reqzmq.connect('tcp://localhost:5556');
+          subzmq.connect('tcp://localhost:5555');
```
Comment (siddharthbhagwan) [inline app/redux/store.js]: Same issue as below.

File: app/components/toolbelt.js [hunk 3]
```diff
@@ -11,12 +11,16 @@ class Toolbelt extends Component {
...
-        </button>
+      <div className="row">
+        <div className="col-md-2">
+          <Button value="Handshake" onClick={this.props.handshake}/>
+        </div>
+        <div className="col-md-3">
+          <button className={`btn ${btnClass}`} data-status={cpuProfile.status} onClick={this.props.handleCpuSampling}>
+            {btnText}
+            {cpuProfile.status === 'RUNNING' ? <i className="fa fa-spinner fa-spin"></i> : null}
+          </button>
```
Comment (siddharthbhagwan) [inline app/components/toolbelt.js]: Instead of creating constants like `btnClass` and `btnText` locally, could we update the button component itself to handle a classname and button text?

File: app/redux/store.js [hunk 4]
```diff
@@ -31,28 +31,31 @@ const zmqMiddlewareCreator = (zmq, subzmq) => {
...
-          zmq.send(sock.cmd);
+          if (zmq.cmd && zmq.cmd.indexOf('stop') === 0) {
+            subzmq.unsubscribe('');
+          }
+
+          reqzmq.send(zmq.cmd);
           break;
         case 'DISCONNECT':
-          subzmq.unsubscribe();
+          subzmq.unsubscribe('');
```
Comment (siddharthbhagwan) [inline app/redux/store.js]: Any particular reason for adding the `' '` parameter while unsubscribing?

File: app/redux/store.js [hunk 5]
```diff
@@ -11,17 +11,17 @@ const subscriber = ZMQ.socket('sub');
...
 
       console.log('zmqMiddleWare :: ', action);
 
-      const {type, sock, ...rest} = action;
-      console.log('zmqMiddleWare :: ', sock, type, rest);
+      const {type, zmq, ...rest} = action;
+      console.log('zmqMiddleWare :: ', zmq, type, rest);
 
       if (type) {
         next({type, ...rest});
```
Comment (siddharthbhagwan) [inline app/redux/store.js:27]: As discussed with @revathskumar the `if (type) {next({type, ...rest})};` block can go towards the end. Its job is to pass on the data to the next middleware if any. Absence of this block breaks the middleware, however returning `next({type, ...rest})` as above in the case of `return next(action)` doesn't work. 

Also, we are yet to figure out the flow here, i.e, is control passed to the next middleware immediately? If so how is the execution of the remaining code in the current middleware and next middleware handled etc.

But to help with the flow, this block might be better placed towards the end.

File: app/redux/store.js (hunk 1 above)
Comment (revathskumar) [inline app/redux/store.js]: Yes we can move this config file. 

``` json
{
  "zmq": {
     "requester": "localhost:5556",
     "subscriber": "localhost:5555"
  }
}
```

Does this looks good?

File: app/redux/store.js (hunk 4 above)
Comment (revathskumar) [inline app/redux/store.js]: Yes. There is a filter option for subscribing in zmq ( [code](https://github.com/JustinTulloss/zeromq.node/blob/master/lib/index.js#L478-L481) ). we need to use the same filter to unsubscribe as well.

File: app/redux/store.js (hunk 1 above)
Comment (iffyuva) [inline app/redux/store.js]: better to store full url, including scheme.

File: app/redux/reducers/connection.js [hunk 6]
```diff
@@ -30,11 +31,19 @@ const handshake = function (state = {}, action) {
...
+    case CONNECTED: {
+      return Object.assign({}, state, {data: []});
+    }
+    case 'DATA': {
       if (action.from !== 'subscriber') {
-        return state;
+        return Object.assign({}, state, {status: 'STOPPED'});
+      }
+      let data = state.data.splice(0);
+      if (state.status === 'STOPPED') {
```
Comment (siddharthbhagwan) [inline app/redux/reducers/connection.js]: What does this condition signify? Data should be cleared only on restarting profiling. (Which is how its working now anyway)

File: app/redux/store.js (hunk 1 above)
Comment (revathskumar) [inline app/redux/store.js]: Do we have any convention on keeping config files? 
Usually I add `config.js.example` to git and gitignore config.js.
Right now I placed config file in app/ directory. should I need to create a config/ directory in root folder and keep config there?

Comment (revathskumar) [inline app/redux/store.js]: As discussed in standup. gitignore the config.js and keep boilerplate config in config.js.example.
All configs will keep in config folder in project root.

Comment (revathskumar) [inline app/redux/store.js]: Also this change will be raised as another PR.

Comment (revathskumar) [inline app/redux/store.js]: Fixed in #10

File: app/redux/store.js (hunk 2 above)
Comment (revathskumar) [inline app/redux/store.js]: Fixed in #10
